CMake/unit_test.cmake

```diff
@@ -18,8 +20,13 @@ function(ADD_UNIT_TEST TESTNAME PROCS THREADS TEST_BINARY)
   endif()
 
   if(TEST_ADDED)
-    set_tests_properties(${TESTNAME} PROPERTIES PROCESSORS ${TOT_PROCS} ENVIRONMENT OMP_NUM_THREADS=${THREADS}
-                                                PROCESSOR_AFFINITY TRUE)
+    set(TEST_ADDITIONAL_ENV
+        $<$<STREQUAL:${ENABLE_SANITIZER},asan>:LSAN_OPTIONS=suppressions=${PROJECT_SOURCE_DIR}/config/sanitizers/lsan.supp>
+    )
+    set_tests_properties(
+      ${TESTNAME} PROPERTIES PROCESSORS ${TOT_PROCS} ENVIRONMENT "OMP_NUM_THREADS=${THREADS};${TEST_ADDITIONAL_ENV}"
+                             PROCESSOR_AFFINITY TRUE)
+    cmake_print_properties(TESTS ${TESTNAME} PROPERTIES ENVIRONMENT)
```

Comment:
I think it can be better to have at least a way to override `LSAN_OPTIONS` instead of being hard-coded internally.

at top level cmakelists.txt around asan.
```
  set(LSAN_OPTIONS "suppressions=${PROJECT_SOURCE_DIR}/config/sanitizers/lsan.supp" CACHE string "cached value of environment variable LSAN_OPTIONS used by ctest")
```
using `APPEND` will be cleaner 
```
set_tests_properties(...) # keep it as it was
if("asan" IN_LIST ENABLE_SANITIZER)
  set_property(TEST ${TESTNAME} APPEND PROPERTY ENVIRONMENT LSAN_OPTIONS=${LSAN_OPTIONS}
endif
```
It will be better if we have this also in "RUN_QMC_APP_NO_COPY" covering all ctest runs.

printing property at every call is an overkill. Let's remove.
This can be revealed by simply running ctest with `-VV`

Reply:
I didn't know that I could use append like that.
Agree on the print, I should have removed that.

Currently it is not possible at least this ccmake to enable more than one sanitizer at a time, is this actually possible?

Reply:
>is this actually possible?

I don't think so. but we won't lose anything by keeping it flexible I assume. Hopefully we can run multiple sanitizer together instead of one at a time.